KIPs/kip-81.md

```diff
@@ -0,0 +1,122 @@
+- Passed: Voting finalized and quorum reached. Executor shall queue the actions before a set timeout of queueTimeout, or 7 days. If the proposal contains no actions, any further interaction is prohibited. Otherwise, the proposal can proceed to Queued and Executed states.
+- Failed: Voting finalized but has not reached quorum. Any further interaction is prohibited.
+- Queued: An executor has queued the actions. An executor cannot trigger the actions while a certain duration of execDelay, or 2 days.
+- Executed: An executor has triggered the actions after the execution delay. The executor shall trigger the actions before a set timeout of execTimeout, or 7 days. The execTimeout starts at the end of execDelay.
+- Expired: Executor has not queued or executed within respective timeouts.
+
+
+![proposal state diagram](../assets/kip-81/proposal_state_diagram.png)
+
+For a proposal to pass, two quorum conditions must be satisfied. The first condition is the turnout. Either the number of voters exceeds 1/3 of all voters, or the sum of casted votes exceeds 1/3 of all eligible votes. The second condition is supported. The yes votes must exceed the half of casted votes.
+The amount of votes are calculated from the amount of stakes KLAY. GC members stake their KLAY by locking in a special smart contract called CnStakingContract. Each member must deploy at least one CnStakingContract and deposit KLAY there. The official addresses of CnStakingContract are managed in another special smart contract called AddressBook.
+Each GC member can appoint a voter account that will cast votes on behalf of the KLAY in its CnStakingContracts. The voter account is necessary because CnStakingContract itself cannot send transactions to cast votes.
+
+![gc voting structure](../assets/kip-81/gc_voting_structure.png)
+  
+Staked balances and voter accounts are tracked in a new special smart contract named StakingTracker. StakingTracker stores the voter list and their voting powers for each voting period. CnStakingContract must notify StakingTracker to recognize any balance changes. Below is an example of StakingTracker contract interface.
+```solidity
+interface IStakingTracker {
+
+  struct Tracker {
+    // this struct is updated only if trackStart <= block.number <= trackEnd.
+    uint64 trackStart;
+    uint64 trackEnd;
+
+    // List of eligible nodes and their staking contracts.
+    // Determined at createTracker() and won’t change.
+    address[] nodeIds;
+    mapping(address => address) stakingToNodeId;
+
+
+    // Balances and voting powers of each nodes.
+    // Updated by notifyStake() and notifyWithdraw().
+    mapping(address => uint256) nodeBalances;
+    mapping(address => uint256) votingPowers;
+    uint256 totalVotes;
+
+
+    // Appointed voters list
+    // Updated by appointVoter().
+    mapping(address => address) nodeIdToVoter;
+    mapping(address => address) voterToNodeId;
+  }
+
+  // Called by Voting at proposal submission, to finalize voter list.
```

Comment:
What is `Voting`? Is this another contract?

Reply:
I will add a list of contracts when we start describing the smart contracts.

Reply:
@kjhman21 Please refer to the "Smart contract specification" section.